src/api/test.rs

```diff
@@ -1492,156 +1492,155 @@ struct LookaheadTestExpectations {
 }
 
 #[test]
-fn lookahead_size_properly_bounded_8() {
-  const LOOKAHEAD_SIZE: usize = 8;
+fn lookahead_size_properly_bounded_7() {
```

Comment:
Why the test changes?

Reply:
oh whoops I think I mixed up two versions of this patch, one which allowed setting to 0 and one which changed the meaning so that 1 was the new minimum. if I allow 0 then the tests shouldn't change, you're right

Reply:
I might squash these 2 commits into 1 because I think it's a bit confusing to follow them separately.